### PR TITLE
 OCP27627

### PR DESCRIPTION
**Pull request by miyadav**
This is to implement the test case for checking - https://bugzilla.redhat.com/show_bug.cgi?id=1796599  
Successful job using the test : https://openshift-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/Runner-v3/72505/console

### CODE REVIEW

**Comment by pruan-rht** [comment]
@miyadav there are conflicts...please fix.

File: features/machine/machine.feature [hunk 1]
```diff
@@ -108,3 +108,24 @@ Feature: Machine features testing
...
+      | o         | jsonpath='{range .items[*]}{.metadata.annotations}{.status.providerStatus.instanceState}{"/\\n"}{end}'|
+
+    And evaluation of `@result[:response].split(":")` is stored in the :console_output_array clipboard
+
+    #Evaluating Result for each line
+    When I repeat the following steps for each :console_output in cb.console_output_array:
+    """
+    And evaluation of `cb.console_output_array.pop.split("]")[0]` is stored in the :provider_status clipboard
+    And evaluation of `cb.console_output_array.pop.split(":").pop.split("]")[1].split("/")[0]` is stored in the :instance_state clipboard
+    Then the expression should be true> cb.provider_status == cb.instance_state
```
Comment (pruan-rht) [inline features/machine/machine.feature]: @miyadav do you know if status.phase == status.providerStatus.instanceState  they look the same to me.  If that's the case we can use existing method and not have to do the string splitting.

Comment (pruan-rht) [reply features/machine/machine.feature]: @miyadav please see #746 for explanation on how to use the new methods.  It should shorten it to only 2 lines of code.  Please let me know if it's unclear

Comment (miyadav) [reply features/machine/machine.feature]: thanks @pruan-rht  , so as I understand  your suggestion is to replace line numbers 116 to 130 with " And evaluation of `BushSlicer::Machine.list(user: admin, project: project('openshift-machine-api'))` is stored in the :machines clipboard
   Then the expression should be true> cb.machines.select {|m| m.instance_state == m.annotation_instance_state }.count == cb.machines.count"  once helper PR is merged?

Comment (jhou1) [reply features/machine/machine.feature]: I think so,  #746  gets merged, your steps can be shorten to the two lines.

Comment (miyadav) [reply features/machine/machine.feature]: thanks @pruan-rht  , going over it will ask when have issues relating to them.

Comment (miyadav) [reply features/machine/machine.feature]: Made changes  and tested working fine locally

**Pull request merged**